test/test_build_redial.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2020 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit Tests for build redial script"""
+import unittest
+from data import build_redial
+
+class TestPreprocessing(unittest.TestCase):
+
+    def test_separate_responses(self):
```

Comment:
It's a bit hard to understand the different cases being tested based on the input (as I'm the least familiar on the dataset). Generally, you can separate the test cases into different tests, e.g.
1) test base case (single user text, single system text)
2) test base case (single system text, single user text)
3) test more complicated case (multiple user text, multiple system text)
4) test multi-turn (multiple user text, multiple system text, ...)
5) test empty input from user or system, so that your codes does not blown up,
...

to give more confidence to others about your processing code.

Will be easier if the test does not depend on an actual file

Reply:
Also if you are going to test with more test cases, then having the input and output in the file is more readable than having to go to check which test data file responds to what.

Reply:
Th other tests I wrote were more that style (on the other branch).  I had this one like this because the separate_responses function expects data in this long format and I figured it was cleaner to put the test cases in a file. I'll change it so that more individual labelled test cases are being tested. I think I'll file that as an issue with the rest of the testing stuff I have to do.